packages/x-ui-admin/src/react/components/RemoteSelect.tsx

```diff
@@ -48,20 +77,38 @@ export function RemoteSelect(props: RemoteSelectProps) {
     <Select
       value={value}
       onChange={(value, option) => {
-        if (Array.isArray(value)) {
-          onChange &&
-            onChange(
-              !idAsString ? value.map((v) => new ObjectId(v)) : value,
-              option
-            );
+        const isArray = Array.isArray(value);
+
+        const defaultOnChange = () => {
+          if (isArray) {
+            onChange &&
+              onChange(
+                !idAsString ? value.map((v) => new ObjectId(v)) : value,
+                option
+              );
+          } else {
+            onChange &&
+              onChange(!idAsString ? new ObjectId(value) : value, option);
+          }
+        };
+
+        if (onSelectedDataChange) {
```

Comment:
`let's just use the value changed` => not sure what you mean. You propose not to have the second argument composed of `isArray, value, onChange: defaultOnChange` ?

Reply:
Yes, we already have the `onChange()`. Its default behavior shouldn't be "opted-out" and it's already called above as I see it, so it's not like you need to call it again? It can be called via `form` (indirectly). And regarding `isArray` when you put the inside `<RemoteSelect>` you know if it's an array or not, right?